### PR TITLE
GFORMS-1153 - Improve performance by limiting data posted to the data…

### PR DESCRIPTION
**Pull request by josef-vlach**
… from the page

Support for "submitMode": "summaryinfoonly" date fields.

### CODE REVIEW

File: test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala [hunk 1]
```diff
@@ -16,588 +16,342 @@
...
+
+  implicit val l: LangADT = LangADT.En
+
+  private val lookupRegistry = new LookupRegistry(Map.empty)
+
+  implicit val smartStringEvaluator: SmartStringEvaluator = new SmartStringEvaluator {
+    override def apply(s: SmartString, markDown: Boolean): String = s.rawValue(LangADT.En)
+  }
+
+  implicit val messages: Messages = Helpers.stubMessages(
```
Comment (girishkamat) [inline test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala:52]: Minor: Could this be moved to a ExampleMessages trait, with all messages for test. I see this being duplicated in DateValidationLogicSpec

Comment (josef-vlach) [reply test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala:52]: That would tight couple those tests. Is that really what we want? In this way every test defined only what it needs, and it even clearly gives a hint what sort of keys are used by a particular test.

Comment (girishkamat) [reply test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala:52]: I know, I had that thought in mind. The messages are just constants and in all cases can represent real messages from the production file. The values themselves have no significance for tests no?

Comment (josef-vlach) [reply test/uk/gov/hmrc/gform/validation/DateValidationSpec.scala:52]: > The messages are just constants and in all cases can represent real messages from the production file

Yes.

Values have significance for test, but they can be anything, they don't need to match real message.

**Review by girishkamat — APPROVED** [review]

**Pull request merged**